### DISCUSSION TITLE
Checking the stability of the model after disturbance

### DISCUSSION THREAD

**Discussion by m-jam20** [CURENT/andes#521] [General]
Hi. Hope you are well.

I would like to check if my model is stable after fault. How can I check it in ANDES? I found two way, but not sure how exactly I should use them to check stability.

The first one is: andes.routines.criteria.deltadelta(delta, diff_limit)

the second one is: TDS.check_criteria()

this is part of my code related to this:

ss.setup()                    
ss.PFlow.run()
ss.TDS.config.tf = tf      
ss.TDS.config.tstep = tstep
ss.TDS.run()
print("------------check_criteria:",ss.TDS.check_criteria())

**Comment by cuihantao** [comment]
This one is the security criteria. Check the definition of stability and security -- a system can be stable but violate security criteria. You will want to inspect the rotor angle curves to tell if the system is stable or not.

**Reply by m-jam20** [reply]
Thanks. Can andes.routines.criteria.deltadelta(delta, diff_limit) check the rotor angle if the system is stable?

**Reply by m-jam20** [reply]
I want to have 4000 samples of different faults for IEEE 14 bus system. Is there any way to code that can detects cases as follows? Should I classify them visually and manually?

Voltage magnitude:
![Image 22-03-2024 at 14 44](https://github.com/CURENT/andes/assets/160192556/4a413939-0527-4822-aa3e-84b4c2c7ab3e)

voltage angels:
![Image 22-03-2024 at 14 44 (1)](https://github.com/CURENT/andes/assets/160192556/389f1ecb-b763-4030-81a6-3f64064bd0bb)
